src/envs/pddl_env.py

```diff
@@ -0,0 +1,442 @@
+        return self._generate_tasks(CFG.num_test_tasks,
+                                    self._pddl_test_problem_generator,
+                                    self._test_rng)
+
+    def _generate_tasks(self, num_tasks: int,
+                        problem_gen: PDDLProblemGenerator,
+                        rng: np.random.Generator) -> List[Task]:
+        tasks = []
+        for pddl_problem_str in problem_gen(num_tasks, rng):
+            task = self._pddl_problem_str_to_task(pddl_problem_str)
+            tasks.append(task)
+        return tasks
+
+    @property
+    def predicates(self) -> Set[Predicate]:
+        return self._predicates
+
+    @property
+    def goal_predicates(self) -> Set[Predicate]:
+        # For now, we assume that all predicates may be included as part of the
+        # goals. This is not important because these environments are not
+        # currently used for predicate invention. If we do want to use these
+        # for predicate invention in the future, we can revisit this, and
+        # try to automatically detect which predicates appear in problem goals.
+        return self._predicates
+
+    @property
+    def types(self) -> Set[Type]:
+        return self._types
+
+    @property
+    def options(self) -> Set[ParameterizedOption]:
+        # The parameterized options are 1:1 with the STRIPS operators.
+        return {
+            self._strips_operator_to_parameterized_option(op)
+            for op in self._strips_operators
+        }
+
+    @property
+    def action_space(self) -> Box:
+        # See class docstring for explanation.
+        num_ops = len(self._strips_operators)
+        max_arity = max(len(op.parameters) for op in self._strips_operators)
+        lb = np.array([0.0 for _ in range(max_arity + 1)], dtype=np.float32)
+        ub = np.array([num_ops - 1.0] + [np.inf for _ in range(max_arity)],
+                      dtype=np.float32)
+        return Box(lb, ub, dtype=np.float32)
+
+    def _action_to_ground_strips_op(
+            self, action: Action,
+            ordered_objects: List[Object]) -> _GroundSTRIPSOperator:
+        action_arr = action.arr
+        assert all(float(a).is_integer() for a in action_arr)
+        op_idx = int(action_arr[0])
+        op = self._ordered_strips_operators[op_idx]
+        op_arity = len(op.parameters)
+        obj_idxs = [int(i) for i in action_arr[1:op_arity + 1]]
+        objs = tuple(ordered_objects[i] for i in obj_idxs)
+        return op.ground(objs)
+
+    def render_state(self,
+                     state: State,
+                     task: Task,
+                     action: Optional[Action] = None,
+                     caption: Optional[str] = None) -> List[Image]:
+        raise NotImplementedError("Render not implemented for PDDLEnv.")
+
+    def _pddl_problem_str_to_task(self, pddl_problem_str: str) -> Task:
+        # Let pyperplan do most of the heavy lifting.
+        # Pyperplan needs the domain to parse the problem.
```

Comment:
let's cache the domain part of this out in the init? you can make your call to _parse_pddl_domain return the pyperplan_domain in addition to current return values